app/src/test/java/org/mozilla/fenix/settings/quicksettings/WebsiteInfoViewTest.kt

```diff
@@ -60,4 +64,63 @@ class WebsiteInfoViewTest {
 
         assertEquals("Insecure Connection", view.securityInfo.text)
     }
+
+    @Test
+    fun `WHEN updating on detailed mode THEN bind the certificate, title and back button listener`() {
+        val view = spyk(WebsiteInfoView(FrameLayout(testContext), icons, interactor, isDetailsMode = true))
+
+        view.update(WebsiteInfoState(
+            websiteUrl = "https://mozilla.org",
+            websiteTitle = "Mozilla",
+            websiteSecurityUiValues = WebsiteSecurityUiValues.INSECURE,
+            certificateName = "Certificate"
+        ))
+
+        verify {
+            view.bindCertificateName("Certificate")
+            view.bindTitle("Mozilla")
+            view.bindBackButtonListener()
+        }
+    }
+
+    @Test
+    fun `WHEN updating on not detailed mode THEN only connection details listener should be binded`() {
+        val view = spyk(WebsiteInfoView(FrameLayout(testContext), icons, interactor, isDetailsMode = false))
+
+        view.update(WebsiteInfoState(
+            websiteUrl = "https://mozilla.org",
+            websiteTitle = "Mozilla",
+            websiteSecurityUiValues = WebsiteSecurityUiValues.INSECURE,
+            certificateName = "Certificate"
+        ))
+
+        verify(exactly = 0) {
+            view.bindCertificateName("Certificate")
+            view.bindTitle("Mozilla")
+            view.bindBackButtonListener()
+        }
+
+        verify {
+            view.bindConnectionDetailsListener()
+        }
+    }
+
+    @Test
+    fun `WHEN rendering THEN use the correct layout`() {
+        val normalView = spyk(
+            WebsiteInfoView(
+                FrameLayout(testContext),
+                icons,
+                interactor,
+                isDetailsMode = false
+            )
+        )
+
+        assertEquals(R.layout.quicksettings_website_info, normalView.layoutId)
+
+        val detailedView =
+            spyk(WebsiteInfoView(FrameLayout(testContext), icons, interactor, isDetailsMode = true))
```

Comment:
Do we need the `spyk()` here?

Reply:
Did we need the spky() for all these views after all? I am mostly wondering since they aren't being used in any `verify { }`.